### PR TITLE
Add examples showing how to use `result` for all tasks

### LINKED ISSUE #19281: After YOLO11 OBB finetuning train images return None boxes
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi!
I am finetuning YOLO OBB on a small (around 100 images) dataset on 2 new classes (locks, keyholes). I am using a small amount of images just to understand if YOLO can overfit on those data. Since I know that the number of images can be low, I am testing the approach on a train image. 

It seems that the model converge, but when I test on a train image I get ```None``` boxes. Here's my script:
```python 
from ultralytics import YOLO
import wandb


# Load a pretrained YOLOv9 model
model = YOLO("yolo11n-obb.pt")
model_path = "yolo_rot_lock1_2.pt"

exp_name = "yolorotations"
wandb.login(key=my_key)
wandb.init(
    project="my_project",
    entity='my_entity',
    group='yolo_training',
    name=exp_name)
# Train the model on your dataset
results = model.train(
    data="./data.yaml",
    epochs=100,
    project="yolo_training",  # wandb project name
    name="yolo11rot"     # wandb run name
)
model = YOLO('./yolo_training/yolo11rot/weights/best.pt')
results = model.predict(source='./rotation_lock1/data/images/train/frame_00000140_left_00.png', conf=0.25)

print(results[0].boxes)
model.save(model_path)
wandb.log_model(model_path)
wandb.finish()
```
I already tried with a lower conf, e.g. ```conf=0.1```. The same approach worked for non-OBB YOLO, with the same images but with non-rotated boxes, so less precise boxes.
I thought that this problem could arise from the labels, but I followed the documentation and the results in the training were good (as it is possible to see in the attached image), so I do not think it was a problem of labels. Moreover, I am not specifying the ```imgsz``` parameter, so I am keeping the default one. What do you think could be the problem? 
Thanks

Image

### Additional

_No response_

### PR DESCRIPTION
**Pull request by Y-T-G**
Closes #19281 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary  
Enhanced documentation for accessing and interpreting prediction results across detection, segmentation, pose estimation, and oriented bounding box tasks, improving clarity for developers. 📚✨  

### 📊 Key Changes  
- Added detailed examples for **results object** handling in detection tasks, including box coordinates (xywh, xyxy), class names, and confidence scores.  
- Expanded documentation for **oriented bounding boxes (OBBs)**, showing how to retrieve center, dimensions, rotation (xywhr), polygon data, and associated metadata.  
- Updated **pose estimation task docs**, clarifying how to extract keypoint coordinates, normalized keypoints, and visibility data.  
- Improved segmentation task examples, explaining mask extraction in polygon and matrix formats alongside normalization options.  

### 🎯 Purpose & Impact  
- **Purpose**: Provide developers with clear, actionable guidance on working with prediction outputs, reducing confusion and speeding up integration. 🛠️  
- **Impact**: Users can now easily access detailed result attributes for bounding boxes, masks, keypoints, and OBBs, streamlining use cases in object detection, segmentation, and pose estimation tasks. 🚀  
- **Developer Benefit**: Saves time on experimenting with results object structure, making workflows more efficient and beginner-friendly. 🕒💡

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your contribution, please review the following checklist:  

- ✅ **Define a Purpose**: Thank you for clearly explaining the purpose of your PR! Linking it to Issue #19281 helps us understand the context and importance of your changes—great work!  
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.  
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address these issues.  
- ✅ **Update Documentation**: You've made relevant updates to our [documentation](https://docs.ultralytics.com/) (thank you for that)! Please double-check all new examples are correctly formatted and consistent with our style guidelines.  
- ✅ **Add Tests**: If applicable, ensure unit tests are included or updated to cover your changes, and confirm that all tests are passing.  
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.  

If any additional changes are necessary, don’t hesitate to update your PR. For further guidance, check out our [Contributing Guide](https://docs.ultralytics.com/help/contributing/).  

⚠️ In case your PR introduces a bug fix or core functionality change, we kindly request a **Minimum Reproducible Example (MRE)** showcasing the issue and your solution, where applicable, so our team can validate its effectiveness.  

We appreciate your efforts and look forward to collaborating further with you! An Ultralytics engineer will review and assist with your PR soon. 🎉🚀

**Comment by Laughing-q** [comment]
@Y-T-G oh thanks for the PR! I don't see we have such comment in other tasks page as well, perhaps we could add the same for other tasks?
pic-250217-1948-56

**Comment by UltralyticsAssistant** [comment]
🎉 Huge thanks to @Y-T-G and @Laughing-q for this fantastic enhancement to our docs! Your detailed examples and clarity are a true gift to the developer community. 🚀  

As Leonardo da Vinci once said, *"Simplicity is the ultimate sophistication."* Your work embodies this spirit by turning complex result interpretation into a user-friendly guide, empowering others to explore detection, segmentation, pose estimation, and OBB tasks with ease.  

Your contributions not only save developers precious time but also inspire deeper engagement with the tools we provide. 🌟 Grateful for your dedication and attention to detail—Ultralytics is better because of you! 🙌 #Teamwork
